internal/command/clistate/local_state.go

```diff
@@ -68,6 +68,15 @@ func (s *LocalState) State() *tofu.State {
 //
 // StateWriter impl.
 func (s *LocalState) WriteState(state *tofu.State) error {
+	err := s.writeState(state)
+	if err != nil {
+		return err
+	}
+
+	// Sync after write
+	return s.stateFileOut.Sync()
```

Comment:
Shouldn't `Sync` be under the the mutex lock too? This PR is changing that behaviour. Is this on purpose?